### PR TITLE
Update how_to_create_badge.md

### PR DESCRIPTION
**Pull request by kochhar**
Following changes made:

* Described contextual details which enable defining request params
* Contextualised the request/response bodies w.r.t the scenario defined
* Changed prerequisites to be specific to badging -- i.e. badging requires users, orgs and roles
* Updated formatting to specify javascript formatting.

Note that indentation in markdown needs to be done using spaces (not tab characters). Have updated all the code blocks to indent using spaces only.

### CODE REVIEW

File: developer-docs/how-to-guide/how_to_create_badge.md [hunk 1]
```diff
@@ -16,178 +16,198 @@ This document explains the procedure of creating and awarding badges.
...
+
+In our scenario, XYZ Corp Certifications decides that it will issue a badge called Basic Alphabet Expert to employees who complete their basic alphabet training. The criteria for awarding the badge is that the employee can list all the letters of the alphabet. This class of badges will be issued by the users who have the `BADGE_ISSUER` role. The graphic design unit at XYZ Corp has designed a cool emblem for this badge which is present on the administrator's computer.
+
+We will use these details to construct the request body for creating badge class.
 
 #### Request Body
 
+The organisation administrator will execute a request with the following parameter values.
+
 Parameter that are submitted through form ```application/x-www-form-urlencoded```, multipart, formdata, or both are usually used as the content type of the request
```
Comment (kochhar) [inline developer-docs/how-to-guide/how_to_create_badge.md]: @Sunitashelar @vrayulu L117 does not make sense in context. Please fix.

File: developer-docs/how-to-guide/how_to_create_badge.md [hunk 2]
```diff
@@ -16,178 +16,198 @@ This document explains the procedure of creating and awarding badges.
...
-	issuerId:issuerslug-66
-	name: Padma
-	description: Badge Issuer 
-	criteria: Should be created by org admin
-	rootOrgId: 0124758459210711040
-	type: user
-	subtype: badge
-	roles: BADGE_ISSUER
-	images : C:\Users\Pictures\badge_logo.png
+```
```
Comment (kochhar) [inline developer-docs/how-to-guide/how_to_create_badge.md]: L119-128 which specify contextual parameters need to be converted into the appropriate cURL request. @Sunitashelar please get correctly formatted version from developer. 

cc: @vrayulu @madhucr

File: developer-docs/how-to-guide/how_to_create_badge.md [hunk 3]
```diff
@@ -16,178 +16,198 @@ This document explains the procedure of creating and awarding badges.
...
-2. The badge ID created is required to [Award the badge](/how_to_create_badge#awarding-the-badge) to a user
+1. The badge ID created is required to [Award the badge](/how_to_create_badge#awarding-the-badge) to a user
 
 ### Awarding the Badge
 
-Once the badge class is created, the organization administrator issues the badge to employees as they finish the course. The badge Assertion API is associated with fetching and listing the assertions of a badge. 
+Once the badge class is created, any user with the `BADGE_ISSUER` role referenced when creating the badge class issues the badge to employees as they finish the course. The badge Assertion API is associated with fetching and listing the assertions of a badge. 
 
->Note: You would have saved the issuer ID received after creating the issuer and the badge ID received after creating the badge class. These IDs are required for awarding the badge.
+> Note: You would have saved the issuer ID received after creating the issuer and the badge ID received after creating the badge class. These IDs are required for awarding the badge.
```
Comment (kochhar) [inline developer-docs/how-to-guide/how_to_create_badge.md]: @Sunitashelar will also need authorization credentials for the user awarding the badge which is missing from the doc. Please connect with the developer to add these details

cc: @vrayulu @madhucr

File: developer-docs/how-to-guide/how_to_create_badge.md (hunk 2 above)
Comment (kochhar) [inline developer-docs/how-to-guide/how_to_create_badge.md]: L119-128 should be converted into the properly formatted cURL request. @Sunitashelar please connect with developer to create the correct request.

cc: @vrayulu @madhucr

File: developer-docs/how-to-guide/how_to_create_badge.md (hunk 1 above)
Comment (kochhar) [inline developer-docs/how-to-guide/how_to_create_badge.md]: L117 does not comprehend in the context. Please revise or delete.

File: developer-docs/how-to-guide/how_to_create_badge.md [hunk 4]
```diff
@@ -16,178 +16,198 @@ This document explains the procedure of creating and awarding badges.
 Sunbird also provides the ability to badge content. While this example shows how to award a badge to a user, the same procedure can be followed to award a badge to content.
  
 ## Prerequisites
+You will to complete some prerequisite steps before you can create badges. You need to 
+1. Get an API Key for accessing[Sunbird APIs](http://www.sunbird.org/apis/). To create an API key, please refer to [How to generate a Sunbird API key]().
```
Comment (kochhar) [inline developer-docs/how-to-guide/how_to_create_badge.md]: xref required: How to generate a Sunbird API key.

File: developer-docs/how-to-guide/how_to_create_badge.md [hunk 5]
```diff
@@ -16,178 +16,198 @@ This document explains the procedure of creating and awarding badges.
...
 
-1. The [API Key for access](http://www.sunbird.org/apis/) and basic authentication
+2. Create users and organisations for those involved in the badge awarding process. This includes the badge recipient and the user who awards the badge. See [How to create users & organisations]() for details of how users and organisations are created and linked.
```
Comment (kochhar) [inline developer-docs/how-to-guide/how_to_create_badge.md]: xref required: How to create users and organisations

File: developer-docs/how-to-guide/how_to_create_badge.md [hunk 6]
```diff
@@ -16,178 +16,198 @@ This document explains the procedure of creating and awarding badges.
...
 
-2. An API client to make API calls. For example use Postman refer [Using Postman](developer-docs/how-to-guide/how_to_use_postman)
+3. Assign roles to specific users who can award badges. For details of role assignment see [How to assign roles]()
```
Comment (kochhar) [inline developer-docs/how-to-guide/how_to_create_badge.md]: xref required: How to assign roles

**Comment by Sunitashelar** [comment]
@kochhar @bvinayakumar some language edits done. missing Xrefs have not been updated. will do so in the course of the day.

**Pull request merged**